### PR TITLE
[New Rule] Potential Persistence via Atom Init Script Modification

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/905

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/macos/persistence_via_atom_init_file_modification.toml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+[metadata]
+creation_date = "2021/01/21"
+maturity = "production"
+updated_date = "2021/01/21"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the modification of the Atom desktop text editor Init File. Adversaries may add malicious JavaScript code to
+init.coffee that will be executed upon Atom application opening.
```
Comment (peasead) [inline rules/macos/persistence_via_atom_init_file_modification.toml:10]:
```suggestion
Identifies the modification of the Atom desktop text editor Init File. Adversaries may add malicious JavaScript code to the
init.coffee file that will be executed upon the Atom application opening.
```

File: rules/macos/persistence_via_atom_init_file_modification.toml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+]
+risk_score = 21
+rule_id = "b4449455-f986-4b5a-82ed-e36b129331f7"
+severity = "low"
+tags = ["Elastic", "Host", "macOS", "Threat Detection", "Persistence"]
+type = "query"
+
+query = '''
+event.category:"file" and not event.type:"deletion" and
+ file.path:/Users/*/.atom/init.coffee and not process.name:(Atom or xpcproxy) and not user.name:root
```
Comment (peasead) [inline rules/macos/persistence_via_atom_init_file_modification.toml:29]: What're your thoughts on this?
```
 file.path:/Users/*/.atom/init.coffee and not process.name:(Atom or xpcproxy) and not user.id>=501
```

Comment (Samirbous) [reply rules/macos/persistence_via_atom_init_file_modification.toml:29]: want to exclude only root user (services), admin account with 0 as user.id post sudo is worth catching

File: rules/macos/persistence_via_atom_init_file_modification.toml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+[metadata]
+creation_date = "2021/01/21"
+maturity = "production"
+updated_date = "2021/01/21"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the modification of the Atom desktop text editor Init File. Adversaries may add malicious JavaScript code to the
+init.coffee file that will be executed upon the Atom application opening.
```
Comment (bm11100) [inline rules/macos/persistence_via_atom_init_file_modification.toml:10]:
```suggestion
Identifies modifications to the Atom desktop text editor Init File. Adversaries may add malicious JavaScript code to the
init.coffee file that will be executed upon the Atom application opening.
```

**Review by bm11100 — APPROVED** [review]

**Review by threat-punter — APPROVED** [review]

**Pull request merged**